scripts/globals/abilities/dark_shot.lua

```diff
@@ -72,7 +72,7 @@ ability_object.onUseAbility = function(player, target, ability)
         ability:setMsg(xi.msg.basic.JA_NO_EFFECT_2)
     end
 
-    local del = player:delItem(2183, 1) or player:delItem(2974, 1)
+    local del = player:delItem(2183, 1, 0) or player:delItem(2974, 1, 0)
     target:updateClaim(player)
```

Comment:
Third argument should be optional, is it breaking without?
`bool CLuaBaseEntity::delItem(uint16 itemID, uint32 quantity, sol::object const& containerID)`

sol::object should accept nil

Reply:
It seems to break on nils
```
onUseAbility: stack index -1, expected number, received nil: not a numeric type 
stack traceback:
	[C]: in function 'delItem'
	./scripts/globals/abilities/dark_shot.lua:75: in function <./scripts/globals/abilities/dark_shot.lua:24>
```

Reply:
it shouldn't even be in a variable. its not used and its exactuation is rigth there. it shoudl just be
```lua
if not player:delItem(2183, 1) then
    player:delItem(2974, 1)
end
```
and if its breaking on nil we have a bigger than script problem involved

Reply:
Going to try to repro this one on my end.  Doesn't make sense since you're passing in two raw ints which should be fine for that function.

Reply:
> 
> 
> it shouldn't even be in a variable. its not used and its exactuation is rigth there. it shoudl just be
> 
> ```lua
> if not player:delItem(2183, 1) then
>     player:delItem(2974, 1)
> end
> ```
> 
> and if its breaking on nil we have a bigger than script problem involved

Well that I agree with I just quickly was only focused on what was broken, its not the return its the methodcall itself

Reply:
just struck me as very very weird.

Reply:
This one's probably on me.  delItem still tried to convert the optional argument to uint32 instead of using the value that provides a bit if sanity.  PR up to fix

Reply:
remove the changes to the element_shot  #files see pr #205